### PR TITLE
[AN-229] Add batch workspace setting

### PR DESCRIPTION
**Pull request by lucymcnatt**
### What
- Add GCP Batch workspace setting option under a closed feature flag
- Only users added to the 'batch-preview' group will be able to enable the setting
- If enabled, cromwell will run workflows on GCP Batch

image
image

### Dependencies
Workspace setting name from https://github.com/broadinstitute/firecloud-orchestration/pull/1503/files

### Testing strategy

- Create private group 'batch-preview'
- Enable the batch setting in a workspace
- Check other user not in the group cannot enable the batch setting
- Run a workflow, check that it ran on GCP batch
Proof of success
image

- [ ]

### CODE REVIEW

File: src/libs/feature-previews-config.ts [hunk 1]
```diff
@@ -150,6 +151,15 @@ const featurePreviewsConfig: readonly FeaturePreview[] = [
     )}`,
     lastUpdated: '12/12/2024',
   },
+  {
+    id: GCP_BATCH,
+    title: 'Run workflows on GCP Batch',
+    description:
+      'Enabling this feature allows workflows to be processed by the GCP Batch service, rather than the former Cloud Life Sciences API.',
```
Comment (jgainerdewar) [inline src/libs/feature-previews-config.ts]: Might want to add a little more context here, like, "All Terra workflows will be transitioned to GCP Batch by mid-2025."

File: src/libs/feature-previews-config.ts [hunk 2]
```diff
@@ -150,6 +151,15 @@ const featurePreviewsConfig: readonly FeaturePreview[] = [
...
+    groups: ['preview-batch'],
```
Comment (kevinmarete) [inline src/libs/feature-previews-config.ts:159]: It seems that I can turn on this feature without being in the `preview-batch` group.


https://github.com/user-attachments/assets/15a92e1d-273c-45c7-843b-aebc40ca02ae

Comment (lucymcnatt) [reply src/libs/feature-previews-config.ts:159]: yeah all feature-previews are visible in every environment except prod
https://github.com/DataBiosphere/terra-ui/blob/f2e39d644cfa2e68b5eab8eb349d9e365b61c9ed/src/libs/feature-previews.js#L31

so this is expected behavior

**Review by kevinmarete — APPROVED** [review]
Looks good to me!

**Review by jgainerdewar — COMMENTED** [review]
Code LGTM! As part of your testing, did you confirm that non-Batch workspaces are still working normally, with and without the feature flag?

**Comment by lucymcnatt** [comment]
> Code LGTM! As part of your testing, did you confirm that non-Batch workspaces are still working normally, with and without the feature flag?

Yup! Workflows don't run on batch when the flag is turned off (or never turned on)

**Review by jgainerdewar — APPROVED** [review]

**Pull request merged**